In [1]:
import pandas as pd
import glob
import os

In [2]:
# get all JSON files all schools in Malang city
files = glob.glob('../data/dapo-kemdikdasmen/Jawa-Timur/Kota Malang/*/*.json')

In [3]:
temp_df_malang_city = []

for f in files:
    df = pd.read_json(f)
    filename = os.path.splitext(os.path.basename(f))[0]
    df.insert(0, 'sekolah', filename)
    temp_df_malang_city.append(df)

df_malang_city = pd.concat(temp_df_malang_city, ignore_index=True)
df_malang_city

sekolah  rombel  guru_kelas  \
0                             SDLB SUMBER DHARMA       7           6   
1                                   SLB SATHITEN       7           2   
2                           SLB YPAC KOTA MALANG      13          12   
3                                 SLBS BCG IDAYU      11           6   
4                        SMALB BCG SUMBER DHARMA       0           0   
5                           SMP LB SUMBER DHARMA       0           0   
6     SLB A,B,D NEGERI KEDUNGKANDANG KOTA MALANG      32          28   
7   SLB C AUTIS NEGERI KEDUNGKANDANG KOTA MALANG      21          20   
8                    SLB YAYASAN PUTRA PANCASILA      13          12   
9                              SLB-B YPTB MALANG      14          10   
10              SLB AUTIS LABORATORIUM UM MALANG      18          11   
11                        SLB AUTISME RIVER KIDS      27          16   
12                         SLB PUTRA JAYA MALANG      14          13   
13                              SLB BHAKTI LUHUR      47          44   

    guru_matematika  guru_bahasa_indonesia  guru_bahasa_inggris  \
0                 0                      0                    0   
1                 0                      0                    1   
2                 0                      0                    0   
3                 0                      0                    0   
4                 0                      0                    0   
5                 0                      0                    0   
6                 0                      0                    0   
7                 0                      0                    0   
8                 0                      0                    0   
9                 0                      0                    0   
10                0                      0                    0   
11                0                      0                    0   
12                0                      0                    0   
13                0                      0                    0   

    guru_sejarah_indonesia  guru_pkn  guru_penjaskes  guru_agama_budi_pekerti  \
0                        0         0               0                        0   
1                        0         0               1                        0   
2                        0         0               0                        0   
3                        0         0               0                        0   
4                        0         0               0                        0   
5                        0         0               0                        0   
6                        0         0               0                        0   
7                        0         0               0                        0   
8                        0         9               1                        9   
9                        0         0               0                        1   
10                       0         0               2                        0   
11                       0         0               0                        3   
12                       0         0               0                        0   
13                       0         0               0                        0   

    ...  sumber_air_minum  kecukupan_air_bersih  jamban_kk  tipe_jamban  \
0   ...               NaN                   NaN        NaN          NaN   
1   ...               NaN                   NaN        NaN          NaN   
2   ...               NaN                   NaN        NaN          NaN   
3   ...               NaN                   NaN        NaN          NaN   
4   ...               NaN                   NaN        NaN          NaN   
5   ...               NaN                   NaN        NaN          NaN   
6   ...               NaN                   NaN        NaN          NaN   
7   ...               NaN                   NaN        NaN          NaN   
8   ...               NaN                   NaN        NaN     

---

In [5]:
# get all JSON files all schools in Malang city
files_jatim = glob.glob('../data/dapo-kemdikdasmen/Jawa-Timur/*/*/*.json')

temp_df_jatim = []

for f in files_jatim:
    df = pd.read_json(f)
    filename = os.path.splitext(os.path.basename(f))[0]
    df.insert(0, 'sekolah', filename)
    temp_df_jatim.append(df)

df_jatim = pd.concat(temp_df_jatim, ignore_index=True)
df_jatim

sekolah  rombel  guru_kelas  guru_matematika  \
0            SLB SAMUDRA LAVENDER      13          13                0   
1                        SLB PGRI      11           6                0   
2    SLB NEGERI KELEYAN BANGKALAN      33          21                0   
3        SLB SAMUDRA TERRA ATHENA       6           6                2   
4              SLB PGRI BANGOREJO      13           9                0   
..                            ...     ...         ...              ...   
387       SLB KHUSUS BINA MANDIRI      10           7                0   
388   SDLB-B KARYA MULIA SURABAYA       7           7                0   
389          SLB-B KARYA MULIA IV       5           5                0   
390           SMALB-B KARYA MULIA      12           5                0   
391           SMPLB B KARYA MULIA       8           7                1   

     guru_bahasa_indonesia  guru_bahasa_inggris  guru_sejarah_indonesia  \
0                        0                    0                       0   
1                        0                    0                       0   
2                        0                    0                       0   
3                        0                    0                       0   
4                        0                    0                       0   
..                     ...                  ...                     ...   
387                      0                    0                       0   
388                      0                    0                       0   
389                      0                    0                       0   
390                      1                    0                       0   
391                      1                    1                       0   

     guru_pkn  guru_penjaskes  guru_agama_budi_pekerti  ...  sumber_air_minum  \
0           0               0                        0  ...               NaN   
1           0               0                        0  ...               NaN   
2           0               0                        0  ...               NaN   
3           0               0                        0  ...               NaN   
4           0               0                        1  ...               NaN   
..        ...             ...                      ...  ...               ...   
387         1               0                        0  ...               NaN   
388         7               7                        7  ...               NaN   
389         0               0                        0  ...               NaN   
390         1               0                        0  ...               NaN   
391         1               0                        1  ...               NaN   

     kecukupan_air_bersih  jamban_kk  tipe_jamban  jml_hari_cuci_tangan  \
0                     NaN        NaN          NaN                   NaN   
1                     NaN        NaN          NaN                   NaN   
2                     NaN        NaN          NaN                   NaN   
3                     NaN        NaN          NaN                   NaN   
4                     NaN        NaN          NaN                   NaN   
..                    ...        ...          ...                   ...   
387                   NaN        NaN          NaN                   NaN   
388                   NaN        NaN          NaN                   NaN   
389                   NaN        NaN          NaN                   NaN   
390                   NaN        NaN          NaN                   NaN   
391                   NaN        NaN          NaN                   NaN   

     jml_wastafel  jml_wastafel_rusak  air_mengalir_wastafel  \
0             NaN                 NaN                    NaN   
1             NaN                 NaN                    NaN   
2             NaN                 NaN                    NaN   
3             NaN                 NaN                    NaN   
4             NaN                 NaN     

In [6]:
df_jatim.describe()

rombel  guru_kelas  guru_matematika  guru_bahasa_indonesia  \
count  392.000000  392.000000       392.000000             392.000000   
mean    13.783163    7.969388         0.466837               0.456633   
std      8.292691    6.622760         1.670849               1.655235   
min      0.000000    0.000000         0.000000               0.000000   
25%     10.000000    4.000000         0.000000               0.000000   
50%     12.000000    7.000000         0.000000               0.000000   
75%     16.250000   10.000000         0.000000               0.000000   
max     60.000000   52.000000        26.000000              26.000000   

       guru_bahasa_inggris  guru_sejarah_indonesia    guru_pkn  \
count           392.000000                   392.0  392.000000   
mean              0.505102                     0.0    0.408163   
std               1.665038                     0.0    1.497392   
min               0.000000                     0.0    0.000000   
25%               0.000000                     0.0    0.000000   
50%               0.000000                     0.0    0.000000   
75%               0.000000                     0.0    0.000000   
max              19.000000                     0.0   19.000000   

       guru_penjaskes  guru_agama_budi_pekerti  guru_seni_budaya  ...  \
count      392.000000               392.000000        392.000000  ...   
mean         0.783163                 0.951531          0.158163  ...   
std          2.865515                 2.430665          0.581457  ...   
min          0.000000                 0.000000          0.000000  ...   
25%          0.000000                 0.000000          0.000000  ...   
50%          0.000000                 0.000000          0.000000  ...   
75%          1.000000                 1.000000          0.000000  ...   
max         40.000000                28.000000          5.000000  ...   

       sumber_air_minum  kecukupan_air_bersih  jamban_kk  tipe_jamban  \
count               0.0                   0.0        0.0          0.0   
mean                NaN                   NaN        NaN          NaN   
std                 NaN                   NaN        NaN          NaN   
min                 NaN                   NaN        NaN          NaN   
25%                 NaN                   NaN        NaN          NaN   
50%                 NaN                   NaN        NaN          NaN   
75%                 NaN                   NaN        NaN          NaN   
max                 NaN                   NaN        NaN          NaN   

       jml_hari_cuci_tangan  jml_wastafel  jml_wastafel_rusak  \
count                   0.0           0.0                 0.0   
mean                    NaN           NaN                 NaN   
std                     NaN           NaN                 NaN   
min                     NaN           NaN                 NaN   
25%                     NaN           NaN                 NaN   
50%                     NaN           NaN                 NaN   
75%                     NaN           NaN                 NaN   
max                     NaN           NaN                 NaN   

       air_mengalir_wastafel  memiliki_drainase  menguras_tangki_septik  
count                    0.0                0.0                     0.0  
mean                     NaN                NaN                     NaN  
std                      NaN                NaN                     NaN  
min                      NaN                NaN                     NaN  
25%                      NaN                NaN                     NaN  
50%                      NaN                NaN                     NaN  
75%                      NaN                NaN                     NaN  
max                      NaN                NaN                     NaN  

[8 rows x 96 columns]

In [9]:
print(f'Rata-rata guru per sekolah: {df_jatim['guru_kelas'].mean()}')
print(f'Rata-rata rombel per sekolah: {df_jatim['rombel'].mean()}')

Rata-rata guru per sekolah: 7.969387755102041
Rata-rata rombel per sekolah: 13.783163265306122


Data yang ditemukan:

- Terdapat 392 Sekolah Luar Biasa (SLB) di Jawa Timur
- Secara rata-rata, total guru yang ada di SLB adalah **7.969**
- Secara rata-rata, total rombel (rombongan belajar) yang ada di SLB adalah **13.783**